### PR TITLE
Better worse, faster slower >:D

### PR DESCRIPTION
**Pull request by MetalPizzaCat**
*no* loops this time >:D
image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
 CFLAGS += -g -O0
+CXXFLAGS += -g -O0 -std=c++20
 LDFLAGS += $(CFLAGS)
 
 vpath % src
 
 # .PHONY all
 
-all: calculator
+all: main
```
Comment (wrac4242) [inline Makefile:15]: should be "calculator" instead of main

Comment (wrac4242) [reply Makefile:15]: unless you either remove or delete line 18

File: main.cpp [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+ * @brief Get and pop last two items on the stack
+ *
+ * @param stack
+ * @param a
+ * @param b
+ */
+void getTwoTopItems(std::stack<int> &stack, int &a, int &b)
+{
+    a = stack.top();
+    stack.pop();
```
Comment (wrac4242) [inline main.cpp:27]: could this be condensed down into `a = stack.pop();`?

File: main.cpp [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    b = stack.top();
+    stack.pop();
+}
+
+void loop(std::stack<int> &stack, std::map<std::string, void (*)(std::stack<int> &stack)> &options, std::string &input)
```
Comment (wrac4242) [inline main.cpp:32]: void pointers, please clean up

File: main.cpp [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    options["div"] = [](std::stack<int> &stack)
+    {
+        doMathToStack(stack, [](auto a, auto b)
+                      { return a / b; });
+    };
+
+    options["exit"] = [](std::stack<int> &stack)
+    {
+        //>:D
+        exit(EXIT_SUCCESS);
```
Comment (wrac4242) [inline main.cpp:91]: exits early, eww, replace outer loop with while true then?

**Review by wrac4242 — CHANGES_REQUESTED** [review]

**Comment by wrac4242** [comment]
PR is almost as good as the "looks good" pr that pinged several million people

File: main.cpp (hunk 2 above)
Comment (MetalPizzaCat) [reply main.cpp:27]: no
as per [ documentaion](https://en.cppreference.com/w/cpp/container/stack/pop) pop returns no value.

File: main.cpp (hunk 3 above)
Comment (MetalPizzaCat) [reply main.cpp:32]: That's not a void pointer that's a function pointer
And specifically pointer to a function that returns `void`

File: main.cpp (hunk 4 above)
Comment (MetalPizzaCat) [reply main.cpp:91]: ***no***
loop bad embrace recursion

File: Makefile (hunk 1 above)
Comment (MetalPizzaCat) [reply Makefile:15]: so be it >:(

**Pull request closed without merging**